python/tests/test_seldon_client.py

```diff
@@ -94,6 +94,20 @@ def test_predict_rest_json_data_ambassador(mock_post):
     assert mock_post.call_count == 1
 
 
+@mock.patch("requests.post", side_effect=mocked_requests_post_success_json_data)
+def test_predict_rest_json_data_ambassador_dict_response(mock_post):
+    sc = SeldonClient(
+        deployment_name="mymodel", gateway="ambassador", client_return_type="dict"
+    )
+    response = sc.predict(json_data=JSON_TEST_DATA)
+    json_response = response.response
+    assert "jsonData" in mock_post.call_args[1]["json"]
+    assert mock_post.call_args[1]["json"]["jsonData"] == JSON_TEST_DATA
+    assert response.success is True
+    assert json_response["jsonData"] == JSON_TEST_DATA
```

Comment:
Perhaps it would be good to add an assertion here for having a `dict` type? I think something like this should work:

```python
assert isinstance(json_response, dict)
```

Similarly, we could add assertions on the existing tests for `SeldonMessage`?

PS: I just realised that the `SeldonMessage` ones use `seldon_message_to_json(...)` so perhaps this is not necessary.